libs/indibase/inditelescope.cpp

```diff
@@ -1223,6 +1226,37 @@ bool Telescope::ISNewSwitch(const char *dev, const char *name, ISState *states,
             return true;
         }
 
+        ///////////////////////////////////
+        // Homing
+        ///////////////////////////////////
+        if (HomeSP.isNameMatch(name))
+        {
+            auto onSwitchName = IUFindOnSwitchName(states, names, n);
+            TelescopeHomeAction action = HOME_NONE;
+            if (!strcmp(onSwitchName, HomeSP[HOME_FIND].getName()))
+                action = HOME_FIND;
+            else if (!strcmp(onSwitchName, HomeSP[HOME_SET].getName()))
+                action = HOME_SET;
+            else
+                action = HOME_GO;
```

Comment:
Default value is `HOME_NONE` . There is no need to assign a value that is not in accordance with reality.